### PR TITLE
Feat/reward start date

### PR DESCRIPTION
**Pull request by mttwlkr**
- Add reward start date, reward end date, event start date and pre event percent
- Add first pass at weighted token per block calculations + tests

### CODE REVIEW

File: snapshots/src/lp-helpers.ts
Comment (dylangolow) [inline snapshots/src/lp-helpers.ts:26]: Should we remove this now?

Comment (mttwlkr) [reply snapshots/src/lp-helpers.ts:26]: I don't think we need to. I could see it coming back and that logic is frequently touched. Not worth it right now IMO.

File: snapshots/src/lp-snapshot.ts
Comment (dylangolow) [inline snapshots/src/lp-snapshot.ts:166]: Should we add error handling here? What would happen if it wasn't a number or was null, should we throw an error and exit?

Comment (mttwlkr) [reply snapshots/src/lp-snapshot.ts]: @dylangolow  It will not be a number or null if there is no event. When that happens, we calculate tokens per sample using 100% of the market's amount. However, we do need to add a check for the case when someone adds an event, but does not specify a `preEventPercent`. Currently, it would just allocate 100% of the market amount for the preEvent blocks and another 100% for the event blocks. Added validation and pushed

**Review by dylangolow — APPROVED** [review]
LGTM, let's merge this after snapshots are taken this evening and merged prior to merging this PR

**Comment by mttwlkr** [comment]
> LGTM, let's merge this after snapshots are taken this evening and merged prior to merging this PR

@dylan We are using this PR to calculate snapshots. Should we merge now?

**Pull request merged**
